proto/event.proto

```diff
@@ -50,17 +63,53 @@ message Histogram {
   map<string, string> tags = 5;
 }
 
+message AggregatedSet {
+  string name = 1;
+  repeated string values = 2;
+  google.protobuf.Timestamp timestamp = 6;
+  map<string, string> tags = 7;
+}
+
+message AggregatedDistribution {
+  string name = 1;
+  repeated double values = 2;
+  repeated uint32 sample_rates = 3;
+  google.protobuf.Timestamp timestamp = 6;
+  map<string, string> tags = 7;
+}
```

Comment:
What is this data type?

Reply:
This is a metric type that I am missing in Prometheus. It is actually supported by [CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_PutMetricData.html):

> You can publish either individual data points in the Value field, or **arrays of values and the number of times** each value occurred during the period by using the **Values and Counts** fields in the MetricDatum structure.

And it's similar to what I think [DogStatsD has (in limited preview)](https://docs.datadoghq.com/developers/metrics/metrics_type/?tab=distribution#metric-type-definition):
> Unlike the HISTOGRAM metric type, which aggregates on the Agent side during the flush interval, a DISTRIBUTION metric sends **all the raw data** during a flush interval to Datadog, and aggregations occur server-side

This is a counter to store and transfer packs of values before any aggregation is applied (it could not happen at all, like in CW case). Prometheus-like histograms and summaries, or structs like CloudWatch [Statistic Sets](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/publishingMetrics.html#publishingDataPoints1) could be reduced from this data structure downstream.

Reply:
Please note one important difference from our idea to aggregate Histograms in metric buffer (#976). Slightly different approach is proposed: we accumulate values into Distribution in the buffer, and then aggregate Distribution into Histogram / Summary / etc in the sink service.

Reply:
@lukesteensen I've summarised the ideas around the buffer into this [doc](https://github.com/timberio/vector/pull/1052/commits/629a7a7b9e4115f13f197b8c38cd16fa81a900e7). Please be aware that the code is not there yet.

Reply:
Nice! This looks very useful.